### PR TITLE
create post modal

### PR DESCRIPTION
**Pull request by emilyliublair**
### Description

Closes https://github.com/GTBitsOfGood/focus-ga/issues/10

### Type of change

🆕 Feature

### How to Test

interact with modal, input fields and press create post

### Checklist

- [ ]  Code follows design and style guidelines
- [ ]  Code is commented with doc blocks
- [ ]  Latest code has been rebased from base branch (usually `develop`)
- [ ]  Commits follow guidelines (concise, squashed, etc)
- [ ]  Github issues have been linked in relevant commits
- [ ]  Relevant reviewers (EM) have been assigned to this PR

### CODE REVIEW

File: src/components/CreatePostModal.tsx [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+                    setTitleError(true);
+                } else {
+                    setTitleError(false);
+                }
+                if (postData.content.length == 0) {
+                    setBodyError(true)
+                } else {
+                    setBodyError(false);
+                }
+            }
```
Comment (nathangong) [inline src/components/CreatePostModal.tsx:93]: would recommend moving the validation logic here into a separate function, setting the error and removing true/false depending on whether the validation passes or fails

File: src/components/CreatePostModal.tsx [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+                                {disabilities.map((disability) => (
+                                    <li
+                                    key={disability._id}
+                                    onClick={() => toggleDisability(disability)}
+                                    className={`flex items-center p-2 cursor-pointer rounded-lg py-2 hover:bg-gray-100 h-10`}
+                                    >
+                                    { postData.tags.includes(disability) ? 
+                                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" fill="currentColor" className="size-4 mr-2">
+                                        <path fillRule="evenodd" d="M12.416 3.376a.75.75 0 0 1 .208 1.04l-5 7.5a.75.75 0 0 1-1.154.114l-3-3a.75.75 0 0 1 1.06-1.06l2.353 2.353 4.493-6.74a.75.75 0 0 1 1.04-.207Z" clipRule="evenodd" />
+                                    </svg> : null}
```
Comment (nathangong) [inline src/components/CreatePostModal.tsx:231]: what is this svg for? if it's just for the tag background, would recommend using css styling wherever possible

File: src/components/CreatePostModal.tsx [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+                                    stroke="#7d7e82"
+                                    className="size-6"
+                                    >
+                                    <path
+                                        strokeLinecap="round"
+                                        strokeLinejoin="round"
+                                        d="m19.5 8.25-7.5 7.5-7.5-7.5"
+                                    />
+                                </svg>
+                            </div>
```
Comment (nathangong) [inline src/components/CreatePostModal.tsx:217]: can you use lucide icons instead?

File: src/components/CreatePostModal.tsx [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+        }
+    };
+
+    return (
+        props.isOpen ? 
+        <div className="fixed inset-0 bg-black bg-opacity-50 flex justify-center items-center z-50" onClick={handleBackgroundClick}>
+            <div className="bg-white p-6 rounded-lg shadow-lg w-full max-w-5xl h-full max-h-[70%] relative z-50 flex flex-col justify-between">
+                <div className="flex justify-between items-center mb-2">
+                    <div className="text-black text-xl font-bold">Create New Post</div>
+                    <svg onClick={handleClose} xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" strokeWidth={1.5} stroke="currentColor" className="size-6 cursor-pointer">
```
Comment (nathangong) [inline src/components/CreatePostModal.tsx:144]: use lucide icons here too

File: src/components/CreatePostModal.tsx [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+    isOpen: boolean;
+    openModal: () => void;
+    closeModal: () => void;
+}
+
+interface PostData {
+    title: string;
+    content: string;
+    tags: Disability[];
+}
```
Comment (nathangong) [inline src/components/CreatePostModal.tsx:28]: use type instead of interface here

File: src/components/CreatePostModal.tsx [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+
+export default function CreatePostModal( props: Props ) {
```
Comment (nathangong) [inline src/components/CreatePostModal.tsx:30]: Can you make the modal scrollable on overflow? Noticing that if the height of the screen is too small some elements will appear outside of the modal background

**Review by nathangong — CHANGES_REQUESTED** [review]
Frontend looks great! Had some small pieces of feedback regarding code organization

File: src/components/CreatePostModal.tsx (hunk 2 above)
Comment (emilyliublair) [reply src/components/CreatePostModal.tsx:231]: it's a check mark to indicate that the disability is selected in the dropdown

**Comment by DanielChee** [comment]
- Dropdown button should be entirely clickable
- Description textbox should be entirely clickable
- Modal Backdrop should only close when the mouse fully clicks it

Thank you and everything else looks great UX-wise!!

**Review by nathangong — APPROVED** [review]

**Pull request merged**